tests/slsa_analyzer/git_service/test_gitlab.py

```diff
@@ -41,7 +44,70 @@ def test_construct_clone_url_without_token(repo_url: str) -> None:
 )
 def test_construct_clone_url_with_token(repo_url: str, clone_url: str) -> None:
     """Test if the ``construct_clone_url`` method produces proper clone URLs with the access token."""
-    with mock.patch.dict(os.environ, {"MCN_PUBLIC_GITLAB_TOKEN": "abcxyz"}):
-        gitlab = PublicGitLab()
+    with mock.patch.dict(os.environ, {"MCN_GITLAB_TOKEN": "abcxyz"}):
+        gitlab = PubliclyHostedGitLab()
         gitlab.load_defaults()
         assert gitlab.construct_clone_url(repo_url) == clone_url
+
+
+@pytest.mark.parametrize(
+    ("user_config_input", "repo_url", "clone_url"),
+    [
+        pytest.param(
+            """
+            [git_service.gitlab.self_hosted]
+            domain = internal.gitlab.org
+            """,
+            "https://internal.gitlab.org/owner/repo.git",
+            "https://oauth2:abcxyz@internal.gitlab.org/owner/repo.git",
+            id="Self-hosted GitLab is set in user config.",
+        ),
+        pytest.param(
+            """
+            [git_service.gitlab.self_hosted]
+            domain = internal.gitlab.org
+            """,
+            "https://internal.gitlab.org/owner/repo",
+            "https://oauth2:abcxyz@internal.gitlab.org/owner/repo",
+            id="Self-hosted GitLab is set in user config.",
+        ),
+    ],
+)
+def test_construct_clone_url_for_self_hosted_gitlab(
+    user_config_input: str, repo_url: str, clone_url: str, tmp_path: Path
+) -> None:
+    """Test if the ``construct_clone_url`` method produces proper clone URLs with the access token."""
+    user_config_path = os.path.join(tmp_path, "config.ini")
+    with open(user_config_path, "w", encoding="utf-8") as user_config_file:
+        user_config_file.write(user_config_input)
+    # We don't have to worry about modifying the ``defaults`` object causing test
+    # pollution here, since we reload the ``defaults`` object before every test with the
+    # ``setup_test`` fixture.
```

Comment:
I'm not sure if we need to repeat this comment in multiple places. But it's fine if you want to keep it.

Reply:
I don't think it would hurt. I put it there because I treat each test case individually so if someone having this test case failed doesn't need to go to other places to know about this comment.